packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -1098,6 +1099,11 @@ export class Vpc extends VpcBase {
    */
   public readonly availabilityZones: string[];
 
+  /**
+   * Internet Gateway for the VPC
+   */
+  public readonly internetGatewayId: string = '';
+
```

Comment:
Why default to an empty string? This isn't accurate and might be confusing.
Since the `CfnInternetGateway` is created conditionally, a better signal to users would be to define it like so:

```suggestion
  /**
   * Internet Gateway for the VPC. Note that in case the VPC is configured only with ISOLATED subnets, this attribute will be `undefined`.
   */
  public readonly internetGatewayId?: string;
```

Reply:
Done in c38d4fb, had to tweak some tests that weren't expecting `undefined`, and added the test mentioned in other comments!

Reply:
👍